client/src/components/DataFiles/DataFilesModals/DataFilesCompressModal.js

```diff
@@ -0,0 +1,189 @@
+import PropTypes from 'prop-types';
+import * as yup from 'yup';
+import './DataFilesCompressModal.module.scss';
+
+const makeDate = () => {
+  const date = new Date();
+  const d = new Date(date);
+  let month = `${d.getMonth() + 1}`;
+  let day = `${d.getDate()}`;
+  const year = d.getFullYear();
+
+  if (month.length < 2) month = `0${month}`;
+  if (day.length < 2) day = `0${day}`;
+
+  return [year, month, day].join('-');
+};
+
+const DataFilesCompressForm = ({ singleFile, formRef }) => {
+  const dateTimeStr = makeDate();
+  const initialValues = {
+    filenameDisplay: `${singleFile ? dateTimeStr : ''}`,
+    filetype: '.zip'
+  };
+  const validationSchema = yup.object().shape({
+    filenameDisplay: yup.string().required('The filename is required')
+  });
+
+  return (
+    <Formik
+      innerRef={formRef}
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+    >
+      {({ values, setFieldValue }) => {
+        const handelSelectChange = e => {
+          setFieldValue('filetype', e.target.value);
+        };
+        return (
+          <Form>
+            <FormField label="Filename" name="filenameDisplay" />
+            <FormGroup>
+              <Input
+                type="select"
+                name="filetype"
+                bsSize="sm"
+                onChange={handelSelectChange}
+              >
+                <option value=".zip">zip</option>
+                <option value=".tar.gz">tar.gz</option>
+              </Input>
+            </FormGroup>
+          </Form>
+        );
+      }}
+    </Formik>
+  );
+};
+DataFilesCompressForm.propTypes = {
+  singleFile: PropTypes.bool.isRequired,
+  formRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.instanceOf(Object) })
+  ])
+};
+DataFilesCompressForm.defaultProps = {
+  formRef: null
+};
+
+const DataFilesCompressModal = () => {
+  const history = useHistory();
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const status = useSelector(
+    state => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const params = useSelector(
+    state => state.files.params.FilesListing,
+    shallowEqual
+  );
+
+  const isOpen = useSelector(state => state.files.modals.compress);
+  const selectedFiles = useSelector(
+    ({ files: { selected, listing } }) =>
+      selected.FilesListing.map(i => ({
+        ...listing.FilesListing[i]
+      })),
+    shallowEqual
+  );
+  const selected = useMemo(() => selectedFiles, [isOpen]);
+  const formRef = React.useRef();
+  const toggle = () =>
+    dispatch({
+      type: 'DATA_FILES_TOGGLE_MODAL',
+      payload: { operation: 'compress', props: {} }
+    });
+
+  const onOpened = () => {
+    dispatch({
+      type: 'FETCH_FILES_MODAL',
+      payload: { ...params, section: 'modal' }
+    });
+  };
+
+  const onClosed = () => {
+    dispatch({ type: 'DATA_FILES_MODAL_CLOSE' });
+    if (isString(status)) {
```

Comment:
If `status` is either `null` with the change I proposed, or a string, we can simply check if it's falsy
```suggestion
    if (status) {
```